TestVolumeProvision is flaky

**Which jobs are failing**:

pull-kubernetes-integration

**Which test(s) are failing**:

TestVolumeProvision

Example: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/81330/pull-kubernetes-integration/1161315691412852736/

```
--- FAIL: TestVolumeProvision (12.16s)
    volume_binding_test.go:1149: Provisoning annotaion on PVC volume-schedulingfc108048-dc1f-4999-8323-f3302c9eb3ca/pvc-w-canbind not bahaviors as expected: PVC volume-schedulingfc108048-dc1f-4999-8323-f3302c9eb3ca/pvc-w-canbind not expected to be provisioned, but found selected-node annotation
    volume_binding_test.go:1191: PV pv-w-canbind phase not Bound, got Available
```

**Since when has it been failing**:

Not sure, [triage board](https://storage.googleapis.com/k8s-gubernator/triage/index.html?pr=1&test=TestVolumeProvision) and [test grid](https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-integration&sort-by-flakiness=) aren't displaying any results

/sig storage

--- a/test/integration/volumescheduling/volume_binding_test.go
+++ b/test/integration/volumescheduling/volume_binding_test.go
@@ -762,7 +762,7 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
 	}
 
 	run := func(t *testing.T, test testcaseType) {
-		// Create StorageClasses
+		t.Log("Creating StorageClass")
 		suffix := rand.String(4)
 		classes := map[string]*storagev1.StorageClass{}
 		classes[classImmediate] = makeDynamicProvisionerStorageClass(fmt.Sprintf("immediate-%v", suffix), &modeImmediate, nil)
@@ -783,27 +783,35 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
 				t.Fatalf("Failed to create StorageClass %q: %v", sc.Name, err)
 			}
 		}
-		// Create PVs
+
+		t.Log("Creating PVs")
 		for _, pvConfig := range test.pvs {
 			pv := makePV(pvConfig.name, classes[pvConfig.scName].Name, pvConfig.preboundPVC, config.ns, pvConfig.node)
 			if _, err := config.client.CoreV1().PersistentVolumes().Create(context.TODO(), pv, metav1.CreateOptions{}); err != nil {
 				t.Fatalf("Failed to create PersistentVolume %q: %v", pv.Name, err)
 			}
+			if err := waitForPVPhase(config.client, pvConfig.name, v1.VolumeAvailable); err != nil {
+				t.Fatalf("PersistentVolume %q failed to become available: %v", pvConfig.name, err)
+			}
 		}
 
-		// Create PVCs
+		t.Log("Creating PVCs")
 		for _, pvcConfig := range test.boundPvcs {
 			pvc := makePVC(pvcConfig.name, config.ns, &classes[pvcConfig.scName].Name, pvcConfig.preboundPV)
 			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(context.TODO(), pvc, metav1.CreateOptions{}); err != nil {
 				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
 			}
 		}
+
+		t.Log("Creating unbound PVCs")
 		for _, pvcConfig := range test.unboundPvcs {
 			pvc := makePVC(pvcConfig.name, config.ns, &classes[pvcConfig.scName].Name, pvcConfig.preboundPV)
 			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(context.TODO(), pvc, metav1.CreateOptions{}); err != nil {
 				t.Fatalf("Failed to create PersistentVolumeClaim %q: %v", pvc.Name, err)
 			}
 		}
+
+		t.Log("Creating unbound PVCs which should be dynamically provisioned")
 		for _, pvcConfig := range test.provisionedPvcs {
 			pvc := makePVC(pvcConfig.name, config.ns, &classes[pvcConfig.scName].Name, pvcConfig.preboundPV)
 			if _, err := config.client.CoreV1().PersistentVolumeClaims(config.ns).Create(context.TODO(), pvc, metav1.CreateOptions{}); err != nil {
@@ -811,7 +819,7 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
 			}
 		}
 
-		// Create Pod
+		t.Log("Creating the pod to schedule")
 		if _, err := config.client.CoreV1().Pods(config.ns).Create(context.TODO(), test.pod, metav1.CreateOptions{}); err != nil {
 			t.Fatalf("Failed to create Pod %q: %v", test.pod.Name, err)
 		}
@@ -825,7 +833,7 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
 			}
 		}
 
-		// Validate PVC/PV binding
+		t.Log("Validating PVC/PV binding")
 		for _, pvc := range test.boundPvcs {
 			validatePVCPhase(t, config.client, pvc.name, config.ns, v1.ClaimBound, false)
 		}
@@ -840,6 +848,7 @@ func testVolumeProvision(t *testing.T, storageCapacity bool) {
 		}
 
 		// Force delete objects, but they still may not be immediately removed
+		t.Log("Deleting test objects")
 		deleteTestObjects(config.client, config.ns, deleteOption)
 	}
 

